core/policy/oracle.go

```diff
@@ -0,0 +1,192 @@
+		ID:     policyID,
+		Source: fmt.Sprintf("%v%v", policyURL, policyID),
+	}
+}
+
+// Policies converts given values to list of valid policies
+func (pr *Oracle) Policies(policyIDs []string) []market.AccessPolicy {
+	policies := make([]market.AccessPolicy, len(policyIDs))
+	for i, policyID := range policyIDs {
+		policies[i] = pr.Policy(policyID)
+	}
+	return policies
+}
+
+// SubscribePolicies adds given policies to repository and syncs changes of it's rules from TrustOracle
+func (pr *Oracle) SubscribePolicies(policies []market.AccessPolicy, repository *Repository) error {
+	pr.fetchLock.Lock()
+	defer pr.fetchLock.Unlock()
+
+	subscriptionsNew := make([]policySubscription, len(pr.fetchSubscriptions))
+	copy(subscriptionsNew, pr.fetchSubscriptions)
+
+	for _, policy := range policies {
+		index, exist := pr.getPolicyIndex(subscriptionsNew, policy)
+		if !exist {
+			index = len(subscriptionsNew)
+			subscriptionsNew = append(subscriptionsNew, policySubscription{
+				policy:      policy,
+				subscribers: []*Repository{repository},
+			})
+		}
+
+		var err error
+		if err = pr.fetchPolicyRules(&subscriptionsNew[index]); err != nil {
+			return errors.Wrap(err, "initial fetch failed")
+		}
+	}
+
+	pr.fetchSubscriptions = subscriptionsNew
+	return nil
+}
+
+func (pr *Oracle) getPolicyIndex(policyList []policySubscription, policy market.AccessPolicy) (int, bool) {
+	for index, policyMeta := range policyList {
+		if policyMeta.policy == policy {
+			return index, true
+		}
+	}
+
+	return 0, false
+}
+
+func (pr *Oracle) fetchPolicyRules(subscription *policySubscription) error {
+	req, err := requests.NewGetRequest(subscription.policy.Source, "", nil)
+	if err != nil {
+		return errors.Wrap(err, "failed to create policy request")
+	}
+	req.Header.Add("If-None-Match", subscription.eTag)
+
+	res, err := pr.client.Do(req)
+	if err != nil {
+		return errors.Wrapf(err, "failed fetch policy rule %s", subscription.policy)
+	}
+	defer res.Body.Close()
+
+	httptrace.TraceRequestResponse(req, res)
+
+	if res.StatusCode == http.StatusNotModified {
+		return nil
+	}
+	if err := requests.ParseResponseError(res); err != nil {
+		return errors.Wrapf(err, "failed to fetch policy rule %s", subscription.policy)
+	}
+
+	var rules = market.AccessPolicyRuleSet{}
+	err = requests.ParseResponseJSON(res, &rules)
+	if err != nil {
+		return errors.Wrapf(err, "failed to parse policy rule %s", subscription.policy)
+	}
+	subscription.eTag = res.Header.Get("ETag")
+
+	for _, subscriber := range subscription.subscribers {
+		subscriber.SetPolicyRules(subscription.policy, rules)
+	}
+
+	return nil
+}
+
+func (pr *Oracle) fetchLoop() {
+	for {
+		select {
+		case <-pr.fetchShutdown:
+			return
+		case <-time.After(pr.fetchInterval):
+			pr.fetchLock.Lock()
+
+			subscriptionsActive := make([]policySubscription, len(pr.fetchSubscriptions))
+			copy(subscriptionsActive, pr.fetchSubscriptions)
+
+			for index := range subscriptionsActive {
+				var err error
```

Comment:
You don't need to define var before if it used nowhere: 
```suggestion
		if err := pr.fetchPolicyRules(&subscriptionsActive[index]); err != nil {
```